contracts/protocol/integration/AaveGovernanceV2Adapter.sol

```diff
@@ -0,0 +1,141 @@
+    limitations under the License.
+
+    SPDX-License-Identifier: Apache License, Version 2.0
+*/
+
+pragma solidity 0.6.10;
+pragma experimental "ABIEncoderV2";
+
+
+/**
+ * @title AaveGovernanceV2Adapter
+ * @author Noah Citron
+ *
+ * Governance adapter for Aave governance that returns data for voting
+ */
+contract AaveGovernanceV2Adapter {
+
+    /* ============ Constants ============ */
+
+    // Signature of delegate function
+    string public constant DELEGATE_SIGNATURE = "delegate(address)";
+
+    // Signature of vote function
+    string public constant VOTE_SIGNATURE = "submitVote(uint256,bool)";
+
+    // Signature of propose function
+    string public constant PROPOSE_SIGNATURE = "create(address,address[],uint256[],string[],bytes[],bool[],bytes32)";
+
+    /* ============ State Variables ============ */
+
+    // Address of Aave proto governance contract
+    address public immutable aaveGovernanceV2;
+
+    // Address of the Aaave token
+    address public immutable aaveToken;
+
+    /* ============ Constructor ============ */
+
+    /**
+     * Set state variables
+     *
+     * @param _aaveGovernanceV2    Address of AAVE Governance V2 contract
+     */
+    constructor(address _aaveGovernanceV2, address _aaveToken) public {
+        aaveGovernanceV2 =  _aaveGovernanceV2;
+        aaveToken = _aaveToken;
+    }
+
+    /* ============ External Getter Functions ============ */
+
+    /**
+     * Generates the calldata to vote on a proposal.
+     *
+     * @param _proposalId           ID of the proposal to vote on
+     * @param _support              Boolean indicating whether to support proposal
+     *
+     * @return address              Target contract address
+     * @return uint256              Total quantity of ETH (Set to 0)
+     * @return bytes                Propose calldata
+     */
+    function getVoteCalldata(uint256 _proposalId, bool _support, bytes memory /* _data */) external view returns (address, uint256, bytes memory) {
+        bytes memory callData = abi.encodeWithSignature(VOTE_SIGNATURE, _proposalId, _support);
+        return (aaveGovernanceV2, 0, callData);
+    }
+
+    /**
+     * Generates the calldata to delegate votes to another ETH address. Self and zero address allowed, which is equivalent to registering and revoking in Compound
+     * like governance systems.
+     *
+     * @param _delegatee            Address of the delegatee
+     *
+     * @return address              Target contract address
+     * @return uint256              Total quantity of ETH (Set to 0)
+     * @return bytes                Propose calldata
+     */
+    function getDelegateCalldata(address _delegatee) external view returns (address, uint256, bytes memory) {
+        bytes memory callData = abi.encodeWithSignature(DELEGATE_SIGNATURE, _delegatee);
+        return (aaveToken, 0, callData);
+    }
+
+    /**
+     * Generates the calldata to create a new proposal
```

Comment:
Can add some more description to this function javadoc. Liked the detail jarek added in his PR 

https://github.com/SetProtocol/set-protocol-v2-contracts/pull/16/files#diff-653de1bac666271855c3fd116f2d234543b1ebc8dcba9718be5943f1370fb5a1R111